File: api/read_excel.js
const ExcelJS = require('exceljs');
async function read() {
  const wb = new ExcelJS.Workbook();
  await wb.xlsx.readFile('C:\\Users\\abhis\\OneDrive\\Desktop\\PoornasreeAI\\docs\\Engineers Training.xlsx');
  
  const sheet = wb.getWorksheet('Training Data') || wb.worksheets[0];
  console.log("Columns:", sheet.getRow(1).values);
  
  const tags = new Set();
  sheet.eachRow((row, rowIdx) => {
    if (rowIdx === 1) return;
    const tag = row.values[1];
    if (tag) tags.add(tag);
  });
  console.log("Unique Tags:", Array.from(tags));
}
read().catch(console.error);

File: read_excel.py
import pandas as pd
file_path = r"C:\Users\abhis\OneDrive\Desktop\PoornasreeAI\docs\Engineers Training.xlsx"
try:
    df = pd.read_excel(file_path)
    print("Columns:", list(df.columns))
    if "tag" in df.columns or "tag " in df.columns:
        tag_col = "tag" if "tag" in df.columns else "tag "
        print("Unique tags:", df[tag_col].unique())
    else:
        print("No tag column found. Here are all columns:")
        print(df.columns)
except Exception as e:
    print("Error:", str(e))
